File: February Long 2012/CLMBSTRS.c
#define MAX 1000000007
#include<stdio.h>
int getCount(int Num){
	char str[50];
	int ctr=0;
	itoa(Num,str,2);
	int i=0;
	char ch;
	while((ch=str[i])!='\0'){
		if(ch=='1'){
			ctr++;
		}
		i++;
	}
	return ctr;
}
int main(){
	int T;
	int N,G;
	scanf("%d",&T);
	int arr[100000];
	arr[0]=1;
	arr[1]=2;
	int K=2;
	while(T>0){
		scanf("%d %d",&N,&G);
		if(N<=K){
			if(getCount(arr[N-1])==G){
				printf("CORRECT\n");
			}
			else{
				printf("INCORRECT\n");
			}
		}
		else{
			while(K!=N){
				arr[K]=(arr[K-1]+arr[K-2])%MAX;
				K++;
			}
			K--;
			if(getCount(arr[N-1])==G){
				printf("CORRECT\n");
			}
			else{
				printf("INCORRECT\n");
			}
		}
		T--;
	}
	return 0;
}

File: February Long 2012/CLMBSTRS.java
import java.io.*;
class CLMBSTRS{
	private static int getCount(int Num){
		String str=Integer.toBinaryString(Num);
		int len=str.length();
		int i=0;
		int ctr=0;
		while(i<len){
			if(str.charAt(i)=='1'){
				ctr++;
			}
			i++;
		}
		return ctr;
	}
	public static void main(String args[])throws IOException{
		BufferedReader I=new BufferedReader(new InputStreamReader(System.in));
		int T;
		int N,G;
		int K=2;
		int arr[]=new int[1000000];
		arr[0]=1;
		arr[1]=2;
		String str;
		int pos;
		T=Integer.parseInt(I.readLine());
		while(T>0){
			str=I.readLine();
			pos=str.indexOf(" ");
			N=Integer.parseInt(str.substring(0,pos));
			G=Integer.parseInt(str.substring(pos+1));
			if(N<=K){
				if(getCount(arr[N-1])==G){
					System.out.println("CORRECT");
				}
				else{
					System.out.println("INCORRECT");
				}
			}
			else{
				while(K<(N-1)){
					arr[K]=(arr[K-1]+arr[K-2])%1000000007;
					K++;
				}
				arr[N-1]=(arr[N-2]+arr[N-3])%1000000007;
				if(getCount(arr[N-1])==G){
					System.out.println("CORRECT");
				}
				else{
					System.out.println("INCORRECT");
				}
			}
			T--;
		}
		
	}
}